src/mlpack/methods/neat/genome_impl.hpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * @file genome.hpp
+ * @author Rahul Ganesh Prabhu
+ *
+ * Implementation of the Genome class which represents a genome in the 
+ * population.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_NEAT_GENOME_IMPL_HPP
+#define MLPACK_METHODS_NEAT_GENOME_IMPL_HPP
+
+// In case it hasn't been included already.
+#include "genome.hpp"
+#include "acyclic_net.hpp"
+
+namespace mlpack{
+namespace neat /** NeuroEvolution of Augmenting Topologies */{
+
+// Creates object.
+template <class ActivationFunction>
+Genome<ActivationFunction>::Genome(const size_t inputNodeCount,
+                                   const size_t outputNodeCount,
+                                   ActivationFunction& actFn,
+                                   const double bias,
+                                   const double weightMutationProb,
+                                   const double weightMutationSize,
+                                   const double biasMutationProb,
+                                   const double biasMutationSize,
+                                   const double nodeAdditionProb,
+                                   const double connAdditionProb,
+                                   const bool isAcyclic):
+    inputNodeCount(inputNodeCount),
+    outputNodeCount(outputNodeCount),
+    actFn(actFn),
+    bias(bias),
+    weightMutationProb(weightMutationProb),
+    weightMutationSize(weightMutationSize),
+    biasMutationProb(biasMutationProb),
+    biasMutationSize(biasMutationSize),
+    nodeAdditionProb(nodeAdditionProb),
+    connAdditionProb(connAdditionProb),
+    isAcyclic(isAcyclic)
+{
+  // Create the node gene list.
+  for (size_t i = 0; i <= inputNodeCount + outputNodeCount; i++)
+    nodeGeneList.push_back(i);
+
+  // Create connections and add them to the lists.
+  for (size_t i = 0; i <= inputNodeCount; i++)
+  {
+    for (size_t j = inputNodeCount+1; j <= outputNodeCount + inputNodeCount;
+        j++)
+    {
+      connectionGeneList.emplace_back(ConnectionGene(nextInnovID++, 1, i, j));
+      if (directedGraph.find(i) == directedGraph.end())
+      {
+        directedGraph.emplace(std::piecewise_construct, std::forward_as_tuple
+            (i), std::initializer_list<std::pair<int, ConnectionGene>>{{j,
+            ConnectionGene(nextInnovID++, 1, i, j)}});
+      }
+      else
+        directedGraph[i].emplace(j, ConnectionGene(nextInnovID++, 1, i, j));
+    }
+  }
+}
+
+// Loads input.
+template <class ActivationFunction>
+void Genome<ActivationFunction>::Input(const arma::vec& input)
```

Comment:
I guess we could remove the `Input` function here and just pass the input once we call the `Output` function. Let me know what you think.

Reply:
Makes sense, I'll change the name to `Evaluate` then.